### PR TITLE
Fix warnings from deprecated tidyverse functionality

### PR DESCRIPTION
**Pull request by kylebaron**
This continues the work that @kyleam started in #303 to modify tidyverse usage in `pmtables` to keep up with deprecated functionality. 

- First, change `test` make target to use `test_local()` ... this seems to show the warnings rather than `test_dir()`
- Use `list_flatten()` rather than `flatten_if()`
- When using `mutate` / `across`, pass the function as a purrr lambda

### CODE REVIEW

File: R/discrete_table.R [hunk 1]
```diff
@@ -96,8 +96,8 @@ cat_data <- function(data, cols, by = ".total", panel = by,
       ans <- complete(ans, !!!syms(.groups))
       nstart <- length(.groups)
       nend <- ncol(ans)
-      ans <- mutate(ans, across(nstart+1, replace_na, 0))
-      ans <- mutate(ans, across(seq(nstart+2, nend), replace_na, "0 (0.0)"))
+      ans <- mutate(ans, across(nstart+1, ~ replace_na(.x, 0)))
+      ans <- mutate(ans, across(seq(nstart+2, nend), ~ replace_na(.x, "0 (0.0)")))
```
Comment (kyleam) [inline R/discrete_table.R:100]: The change looks fine, and I see that the `~` purrr shorthand is used elsewhere is this package, so this is just my two cents:  in the package context, I dislike that `~`, compared to `function(...) ...` (or `\(x) ...` for R 4.1 or later),  prevents `R CMD check` from "seeing inside" the closure and warning about things (e.g., undefined variable).

**Review by kyleam — APPROVED** [review]
Thanks, LGTM

File: R/discrete_table.R (hunk 1 above)
Comment (kylebaron) [reply R/discrete_table.R]: Thanks. Changed `~` to regular function.

File: Makefile [hunk 2]
```diff
@@ -24,7 +24,7 @@ readme:
 
 test:
 	make install
-	Rscript -e 'testthat::test_dir("tests/testthat")'
+	Rscript -e 'testthat::test_local()'
```
[the remark this answers is not in the document]
Comment (kylebaron) [reply Makefile:27]: It looks like passing the package name will let you see warnings; not sure why that is. 

```
test:
	make install
	Rscript -e 'testthat::test_dir("tests/testthat", package = "pmtables")'
```


```
Rscript -e 'testthat::test_dir("tests/testthat", package = "pmtables")'
✔ | F W S  OK | Context
✔ |        24 | test-align
✔ |        10 | test-as_stable [0.1s]
✔ |        23 | as-image [4.0s]
✔ |   1    64 | test-caption [0.2s]
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Warning (test-caption.R:186:3): caption on stable gets passed into preview
`flatten_if()` is deprecated as of rlang 1.1.0.
This warning is displayed once every 8 hours.
Backtrace:
 1. pmtables::st2report(text, dry_run = TRUE, stem = "test1")
      at test-caption.R:186:2
 3. pmtables (local) `<fn>`(...)
 4. rlang::flatten_if(tables, is.list)
 5. rlang:::deprecate_soft("`flatten_if()` is deprecated as of rlang 1.1.0.")
 6. rlang:::.rlang_lifecycle_deprecate_warn0(...)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
✔ |         9 | test-check
✔ |         5 | test-clear_reps
```

File: NAMESPACE [hunk 3]
```diff
@@ -208,7 +208,6 @@ importFrom(rlang,as_string)
 importFrom(rlang,enquo)
 importFrom(rlang,enquos)
 importFrom(rlang,env_clone)
-importFrom(rlang,flatten_if)
```
Comment (kyleam) [inline NAMESPACE:210]: Bah, sorry for not catching the stale namespace my review.

**Review by kyleam — APPROVED** [review]

File: NAMESPACE (hunk 3 above)
Comment (kylebaron) [reply NAMESPACE:210]: @kyleam  Just to clarify: this is what we want, right? ... Drop `flatten_if` / add `list_flatten`? Or something else here?

Comment (kylebaron) [reply NAMESPACE:210]: OIC; didn't realize that came after you started reviewing. Thanks.

**Pull request merged**